sel/src/Sel/SecretKey/AuthenticatedEncryption.hs

```diff
@@ -0,0 +1,208 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+--
+-- Module: Sel.SecretKey.AuthenticatedEncryption
+-- Description: Authenticated Encryption with Poly1305 MAC and XSalsa20
+-- Copyright: (C) Hécate Moonlight 2022
+-- License: BSD-3-Clause
+-- Maintainer: The Haskell Cryptography Group
+-- Portability: GHC only
+module Sel.SecretKey.AuthenticatedEncryption
+  ( -- ** Introduction
+    -- $introduction
+
+    -- ** Usage
+    -- $usage
+    SecretKey
+  , newSecretKey
+  , Nonce
+  , newNonce
+  , Hash
+  , encrypt
+  , decrypt
+  ) where
+
+import Data.ByteString (StrictByteString)
+import qualified Data.ByteString.Unsafe as BS
+import Foreign (ForeignPtr)
+import qualified Foreign
+import Foreign.C (CChar, CUChar, CULLong)
+import GHC.IO.Handle.Text (memcpy)
+import System.IO.Unsafe (unsafeDupablePerformIO)
+
+import Control.Monad (void)
+import Data.Word (Word8)
+import LibSodium.Bindings.Random (randombytesBuf)
+import LibSodium.Bindings.Secretbox (cryptoSecretboxEasy, cryptoSecretboxKeyBytes, cryptoSecretboxKeygen, cryptoSecretboxMACBytes, cryptoSecretboxNonceBytes, cryptoSecretboxOpenEasy)
+import Sel.Internal
+
+-- $introduction
+-- Authenticated Encryption is the action of encrypting a message using a secret key
```

Comment:
Incredibly minor suggestion: `"Authenticated Encryption" uses a secret key along with a single-use number called a "nonce" to encrypt a message.`